### PR TITLE
Feature/4465 move complete reg action to registration workflow

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-4465

#### Changes proposed in this Pull Request:

- Moved complete registration action to registration workflow component
- Refactored nexScreen function logic

#### Screenshots / Screen Recording (if applicable)
- 
Screenshot 2023-07-19 at 2 37 13 PM

#### To Test:

- yarn start:example2
- Click on Debug
- Click on Registration Workflow
- Go through registration workflow till you see the Success screen

#### Any specific feedback you are looking for?

- Please convert #364 to open once this gets merged.

### CODE REVIEW

File: login-workflow/src/new-architecture/components/RegistrationWorkflow/RegistrationWorkflow.tsx [hunk 1]
```diff
@@ -57,22 +63,62 @@ export const RegistrationWorkflow: React.FC<React.PropsWithChildren<Registration
...
                 setCurrentScreen((i) => i - 1);
             }}
             screenData={screenData}
-            updateScreenData={handleScreenNavigation}
+            updateScreenData={updateScreenData}
         >
-            {screens[currentScreen]}
+            {showSuccessScreen ? (
+                <RegistrationSuccessScreen
+                    errorDisplayConfig={{
```
Comment (joebochill) [inline login-workflow/src/new-architecture/components/RegistrationWorkflow/RegistrationWorkflow.tsx:110]: Do we need this? In my head, if we're on the SuccessScreen, there's no possible situation where we would need to display an error.

Comment (surajeaton) [reply login-workflow/src/new-architecture/components/RegistrationWorkflow/RegistrationWorkflow.tsx]: Makes sense.

**Comment by manojleaton** [comment]
Looks good to me!

**Review by manojleaton — APPROVED** [review]

File: login-workflow/src/new-architecture/screens/SuccessScreen/SuccessScreenBase.tsx [hunk 2]
```diff
@@ -15,7 +24,9 @@ export const SuccessScreenBase: React.FC<SuccessScreenProps> = (props) => {
         <WorkflowCard {...cardBaseProps}>
             <WorkflowCardHeader {...headerProps} />
             <WorkflowCardBody>
-                <WorkflowFinishState icon={icon} title={messageTitle} description={message} />
+                <ErrorManager {...errorDisplayConfig}>
+                    <WorkflowFinishState icon={icon} title={messageTitle} description={message} />
+                </ErrorManager>
```
Comment (daileytj) [inline login-workflow/src/new-architecture/screens/SuccessScreen/SuccessScreenBase.tsx:29]: Why is there an error manager in here? This is a success screen... errors should be handled before we even get to this screen.

**Pull request merged**